rust/agama-dbus-server/src/questions.rs

```diff
@@ -260,6 +243,42 @@ impl Questions {
             questions: HashMap::new(),
             connection: connection.to_owned(),
             last_id: 0,
+            answer_strategies: vec![],
+        }
+    }
+
+    /// tries to provide answer to question using answer strategies
+    ///
+    /// What happens under the hood is that it uses answer_strategies vector
+    /// and try to find the first strategy that provides answer. When
+    /// answer is provided, it returns immediately.
+    fn fill_answer(&self, question: &mut GenericQuestion) {
+        for strategy in self.answer_strategies.iter() {
```

Comment:
Just a suggestion for the future (not even for this PR), but I would say that a version using combinators would be easier to read. Something like (not tested):

```rust
        let answer = self
            .answer_strategies
            .iter()
            .find_map(|s| s.answer(question));

        // Note: I did not use `and_then` because it expects the closure to return an Option, and that's not the case.
        if let Some(value) = answer {
            question.answer = value;
        }
```